econml/metalearners.py

```diff
@@ -51,60 +57,53 @@ def fit(self, Y, T, X, inference=None):
         X : array-like, shape (n, d_x)
             Feature vector that captures heterogeneity.
 
-        inference: string, `Inference` instance, or None
+        inference : string, `Inference` instance, or None
             Method for performing inference.  This estimator supports 'bootstrap'
             (or an instance of `BootstrapInference`)
 
         Returns
         -------
-        self: an instance of self.
+        self : an instance of self.
 
         """
         # Check inputs
         Y, T, X, _ = check_inputs(Y, T, X, multi_output_T=False)
-        if not np.array_equal(np.unique(T), [0, 1]):
-            raise ValueError("The treatments array (T) can only contain" +
-                             "0 and 1.")
-        self.controls_model.fit(X[T == 0], Y[T == 0])
-        self.treated_model.fit(X[T == 1], Y[T == 1])
+        T = self._label_encoder.fit_transform(T)
+        self._one_hot_encoder.fit(T.reshape(-1, 1))
+        self._d_t = len(self._label_encoder.classes_)
```

Comment:
I think this should be `len(...) - 1`, but it might not matter too much for this particular subclass since it doesn't support the things that make a distinction (like statsmodels inference).

Reply:
Yes my original thought was d_t represents number of class in multi treatment settings, finally const_marginal_effect will return m*d_y*d_t-1. But you are right in order for consistency with the rest of the library, I will change it accordingly.